Services/CO.CDP.Tenant.WebApi/UseCase/LookupTenantUseCase.cs

```diff
@@ -1,14 +1,23 @@
 using AutoMapper;
+using CO.CDP.Authentication;
 using CO.CDP.Functional;
 using CO.CDP.OrganisationInformation.Persistence;
+using CO.CDP.Tenant.WebApi.Model;
 
 namespace CO.CDP.Tenant.WebApi.UseCase;
 
-public class LookupTenantUseCase(ITenantRepository tenantRepository, IMapper mapper)
-    : IUseCase<string, Model.TenantLookup?>
+public class LookupTenantUseCase(ITenantRepository tenantRepository, IMapper mapper, IClaimService claimService)
+    : IUseCase<Model.TenantLookup?>
 {
-    public async Task<Model.TenantLookup?> Execute(string userUrn)
+    public async Task<Model.TenantLookup?> Execute()
     {
+        var userUrn = claimService.GetUserUrn();
+
+        if (string.IsNullOrEmpty(userUrn))
+        {
+            throw new UnknownTokenException("Cannot find sub or urn from JWT token.");
```

Comment:
Maybe the error could be something like _Ensure the token is valid and contains the necessary claims_

Reply:
The exception name is misleading. What does it mean the token is unknown? In this scenario, the token might or might not be there, but the information we need is missing (urn).

Wouldn't `MissingUserUrnException` better reflect what happened here?

Also, have you considered the ClaimService to throw this exception instead of returning `null`? It would only work if it's never optional though and I'm not so sure about it.

Reply:
Thanks @glgoaco i will update that.